Index: opensbi-89182b257c8798e15e4c685c1af0c2862d528d2a/lib/utils/irqchip/plic.c
===================================================================
--- opensbi-89182b257c8798e15e4c685c1af0c2862d528d2a.orig/lib/utils/irqchip/plic.c
+++ opensbi-89182b257c8798e15e4c685c1af0c2862d528d2a/lib/utils/irqchip/plic.c
@@ -20,6 +20,16 @@
 #define PLIC_ENABLE_STRIDE 0x80
 #define PLIC_CONTEXT_BASE 0x200000
 #define PLIC_CONTEXT_STRIDE 0x1000
+#define PLIC_CTRL 0x01FFFFC
+
+static int plic_enable_deleg(struct plic_data *plic)
+{
+        /* Delegate plic enable into S-mode */
+	volatile void *plic_ctrl = (void *)plic->addr + PLIC_CTRL;
+        writel(0x1,plic_ctrl);
+
+        return 0;
+}
 
 static void plic_set_priority(struct plic_data *plic, u32 source, u32 val)
 {
@@ -82,6 +92,7 @@ int plic_warm_irqchip_init(struct plic_d
 	if (s_cntx_id > -1)
 		plic_set_thresh(plic, s_cntx_id, 0x7);
 
+	plic_enable_deleg(plic);
 	return 0;
 }
 
@@ -96,5 +107,6 @@ int plic_cold_irqchip_init(struct plic_d
 	for (i = 1; i <= plic->num_src; i++)
 		plic_set_priority(plic, i, 1);
 
+	plic_enable_deleg(plic);
 	return 0;
 }
